spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubEmulator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.stream.binder.pubsub;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.rules.ExternalResource;
+
+/**
+ * JUnit rule that starts a Pub/Sub emulator.
+ *
+ * @author João André Martins
+ */
+public class PubSubEmulator extends ExternalResource {
```

Comment:
There is an `AbstractExternalResourceTestSupport` to consider here for extension.

Reply:
Looking at the [source code](https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-test-support-internal/src/main/java/org/springframework/cloud/stream/test/junit/AbstractExternalResourceTestSupport.java), it's not obvious what `AbstractExternalResourceTestSupport` can do for us that we're not already doing here.
What am I missing?

Reply:
Well, at least that helped me to develop similar tests for Kinesis Binder: https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/blob/master/spring-cloud-stream-binder-kinesis/src/test/java/org/springframework/cloud/stream/binder/kinesis/LocalKinesisResource.java.

The same way we go in the Kafka Binder: https://github.com/spring-cloud/spring-cloud-stream-binder-rabbit/blob/master/spring-cloud-stream-binder-rabbit-test-support/src/main/java/org/springframework/cloud/stream/binder/test/junit/rabbit/RabbitTestSupport.java.

Your problem here that you don't expose resource but rely on the `localhost:8085`.
:shrug:

Also we should somehow ensure my environment on Windows without any Cloud SDK.

Thanks

Reply:
These tests don't run by default.
The `GCP_PUBSUB_EMULATOR` envvar needs to be passed to the process so they are triggered.

Reply:
OK! So, why then they fail in Maven? And I guess they are going to fail for me on Windows

Reply:
I need to look into the Maven issue.
Getting the tests to run on Windows is part of this PR scope. I'll add that in a separate commit.

Reply:
Let's never mind about Windows for now!
That may be addressed in the separate issue.
My concern right now is to run project either way and don't fail.
Just to ignore tests on Windows is enough for me.

Thanks